### PR TITLE
Fixing bug where hasMany could not get classname in some cases

### LINKED ISSUE #51: Error when saving record which has many children
When saving a record with many children, when serializing, an error is thrown in the serializeHasMany function of EmberParseAdapter.Serializer line 184.

          "className": child.parseClassName(),

The parseClassName function does not exist on the child record. This seems like an issue - is it supposed to be calling the parseClassName method on the serializer/adapter rather?

I got around this in my code by implementing a parseClassName method on my own model, which just returns the parse class name of that model.

Seems like a fix would be to change this line to call the parseClassName method on the serializer, rather than the non-existent method on the child model. Does that sound OK, or am I getting confused somewhere?

I'll make the fix if my solution sounds fine.

### PR DESCRIPTION
**Pull request by igorsantos07**
This commit fixes #51

### CODE REVIEW

File: addon/serializers/application.js [hunk 1]
```diff
@@ -190,7 +191,7 @@ export default DS.RESTSerializer.extend({
       hasMany.forEach( function( child ) {
         json[key].objects.push({
           '__type'    : 'Pointer',
-          'className' : child.parseClassName(),
+          'className' : serializer.parseClassName( child.typeKey ),
```
Comment (mixonic) [inline addon/serializers/application.js]: I think it may be more appropriate to use the `serializer` of the `child` type instead?

**Comment by mixonic** [comment]
Looking good! Can you add a test for this fix so we know it won't rise up again?

**Comment by QuantumInformation** [comment]
Are the tests made against a real instance of parse or are there plans to?

On Monday, 27 April 2015, Matthew Beale <notifications@github.com> wrote:

> Looking good! Can you add a test for this fix so we know it won't rise up
> again?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/clintjhill/ember-parse-adapter/pull/67#issuecomment-96678681>
> .
>


-- 

Kind regards
Nikos

**Comment by igorsantos07** [comment]
I'm not sure I can help with that... I'm not used to tests on JS :(

**Comment by igorsantos07** [comment]
It seems this PR was ignored in favour of #70, that patches many more issues.

**Pull request closed without merging**